pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -243,6 +253,32 @@ func buildOutboundAutoPassthroughFilterStack(env *model.Environment, node *model
 	return filterstack
 }
 
+// buildRedisFilter builds an outbound Envoy RedisProxy filter.
+// Currently, if multiple clusters are defined, one of them will be picked for
+// configuring the Redis proxy.
+func buildRedisFilter(statPrefix, clusterName string, isXDSMarshalingToAnyEnabled bool) listener.Filter {
+	config := &redis_proxy.RedisProxy{
+		StatPrefix: statPrefix, // redis stats are prefixed with redis.<statPrefix> by Envoy
+		Settings: &redis_proxy.RedisProxy_ConnPoolSettings{
+			OpTimeout: &redisOpTimeout, // TODO: Make this user configurable
+		},
+		PrefixRoutes: redis_proxy.RedisProxy_PrefixRoutes{
+			CatchAllCluster: clusterName,
```

Comment:
What would be the load balancing type for this cluster?

Reply:
It depends, by default it is round robin between all the endpoints of the cluster

Reply:
OK. Should we configure the redis clusters with either MAGLEV https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing/load_balancers.html?highlight=maglev#maglev or RING_HASH to get the full partition based routing from client perspective?  This can be controlled by another config like` PILOT_REDIS_LB` and if not specified can be defaulted to Round Robin as we are doing today.

Reply:
User can set this by DestinationRule

Reply:
Yes. I know. But trying to see if we can make that automatic without user actually creating a destination rule to simplify configuration. If the idea is OK, I can do a follow-up PR once this is merged.

Reply:
It would be better if this can be done automatically.

Reply:
OK. Sure. I will push a PR to do this automatically once this PR is merged.

Reply:
I wouldn't do another flag.. just set this as default MAGLEV for all redis.. dont bother adding flags. Round robin makes little sense for redis